pkgs/os/os.sh

```diff
@@ -59,27 +63,61 @@ _refresh() {
 
 _switch() {
     echo "${1}" > "${CURRENT_FILE}"
+    _refresh
+}
+
+_check_zone() {
+  ZONE=${1}
+  if ! jq -e -r --arg ZONE "${ZONE}" '.[] | select(.providers.fe.domain.name == $ZONE) | .infra_zone_name' "${CAASCAD_ZONES_FILE}" &>/dev/null && 
+    ! jq -e -r --arg ZONE "${ZONE}" '.[$ZONE].infra_zone_name' "${CAASCAD_ZONES_FILE}" &>/dev/null; then
+    echo "Zone or domain name ${ZONE} not found" && exit 1;
+  fi
+}
+
+_get_vault_fqdn() {
+  ZONE="${1}"
+  if [[ $ZONE_NAME =~ ^OCB000.* ]]; then
+      INFRA_ZONE_NAME="$(jq -e -r --arg ZONE "${ZONE}" '.[] | select(.providers.fe.domain.name == $ZONE) | .infra_zone_name' "${CAASCAD_ZONES_FILE}" )"
+      DOMAIN_NAME="$(jq -r --arg ZONE "${INFRA_ZONE_NAME}" '.|to_entries[] | select(.key == $ZONE) | .value.domain_name' "${CAASCAD_ZONES_FILE}" )"
+  else
+      INFRA_ZONE_NAME="$(jq -e -r --arg ZONE "${ZONE}" '.[$ZONE].infra_zone_name' "${CAASCAD_ZONES_FILE}")"
+      DOMAIN_NAME="$(jq -r --arg ZONE "${ZONE}" '.[$ZONE].domain_name' "${CAASCAD_ZONES_FILE}")"
+  fi
+  echo "https://vault.${INFRA_ZONE_NAME}.${DOMAIN_NAME}"
+}
+
+_get_current_zone() {
+  ZONE=$(cat "${CURRENT_FILE}")
+  if [ -z "${ZONE}" ]; then
+    echo "Zone name empty. Use os switch first." >&2
+    exit 1
+  fi
+  echo "${ZONE}"
 }
 
 _parse() {
     if [[ "$#" -eq 0 ]]; then
-        cat "${CURRENT_FILE}";
+        _get_current_zone;
         exit 0;
     fi
     case "$1" in
         os_help|ch)
             _help
             ;;
         print|p)
-            _get_secrets
             _print "${@:2}"
             ;;
+        support-template|st)
+            _get_secrets
+            _support_template
+            ;;
         refresh|r)
             _refresh;
             ;;
         switch|s)
             if [[ "$#" -eq 2 ]]; then
-                _switch "$2"
+                _switch "${2}"
+                _check_zone "${2}"
```

Comment:
put `_check_zone()` inside of functions fetching or setting zone instead ? (`_switch()` & `_get_current_zone()`)

Reply:
done